File: server/api/v1/models/Propiedad.py
class Propiedad:
    def __init__(self, id, ubicacion, habitaciones, fechaPublicacion, lat, longitud, amenities, imagenes):
        self.id = id
        self.ubicacion = ubicacion
        self.habitaciones = habitaciones
        self.fechaPublicacion = fechaPublicacion
        self.lat = lat
        self.longitud = longitud
        self.amenities = amenities
        self.imagenes = imagenes

    
    def tojson(self):
        json = { 
            "id" : self.id, 
            "ubicacion" : self.ubicacion,
            "habitaciones":  self.habitaciones,
            "fechaPublicacion" : self.fechaPublicacion,
            "lat": self.lat,
            "longitud": self.longitud,
            "amenities": self.amenities,
            "imagenes": self.imagenes
        } 
        return json

File: nosqlserver/api/models/propiedad.js
const mongoose = require('mongoose');

const { Schema } = mongoose;

const propiedadSchema = new Schema({
    id: { type: Number, required:true, unique:true},
    ubicacion: String,
    habitaciones: Number,
    fechaPublicacion: Date,
    lat: String,
    longitud: String,
    usuario: {
        rol: Number,
        descripcion: { type: String, enum: [ "Propietario", "Inmobiliaria", "Inquilino" ]}
    },
    precios: Number,
    amenities: [
        {
            descripcion: { type: String, enum: [ "Asador", "Cochera", "Wifi", "Patio", "Balcon", "Ascensor" ]},
            valor: String
        }
    ]
})

const model = mongoose.model("Propiedad", propiedadSchema);

module.exports = model;